### ISSUE TITLE
Graphs on admin page do not populate

### ISSUE THREAD

**Issue by PartisanEntity** [pi-hole/pi-hole#1175]
**In raising this issue, I confirm the following (please check boxes, eg [X]) Failure to fill the template will close your issue:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*
- [x] The issue I'm reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the codebase?:**

1

---

**[BUG | ISSUE] Expected Behaviour:**

When loading the /admin page. The graphs should populate

**[BUG | ISSUE] Actual Behaviour:**

The graphs do not populate.
Tested on Ubuntu LTS Desktop 16.04 in Chromium, Firefox and Google Chrome as well as Windows 7 and IE11.

**[BUG | ISSUE] Steps to reproduce:**

- Install pi-hole on a raspberry pi 3
- Use for a couple days
- Graphs will not populate

**Did you clear the browser cache?:**

Yes on both operating systems and in all 4 browsers.

**(Optional) Debug token generated by `pihole -d`:**

`<m4j4qilnss>`

**Screenshots from all browswers:**

![google-chrome-admin-nologin](https://cloud.githubusercontent.com/assets/4522478/22373714/6016b3ce-e4a3-11e6-9d74-7bd62ad27858.png)
![google-chrome-admin-login](https://cloud.githubusercontent.com/assets/4522478/22373715/601a27a2-e4a3-11e6-9273-d8e79aae7dbe.png)
![firefox-admin-nologin](https://cloud.githubusercontent.com/assets/4522478/22373717/601aa54c-e4a3-11e6-8a42-2d00aadd330a.png)
![firefox-admin-login](https://cloud.githubusercontent.com/assets/4522478/22373716/601a8774-e4a3-11e6-89c6-b9719db38deb.png)
![chromium-admin-nologin](https://cloud.githubusercontent.com/assets/4522478/22373719/601afdee-e4a3-11e6-924f-dc125cd1a695.png)
![chromium-admin-login](https://cloud.githubusercontent.com/assets/4522478/22373718/601ace3c-e4a3-11e6-8d67-954fd647e0ba.png)

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

**Comment by jacobsalmela** [comment]
It looks like your log file is way too big for the Pi to handle.

Can you run 
```
pihole -f
```
which will flush the logs so all of your 1.6 million queries will be lost.  You could backup this file if you really wanted them.

You should also check if the `cron` job is firing, which flushes the log daily.
```
cat /etc/cron.d/pihole
```

**Comment by PartisanEntity** [comment]
Thank you very much for the lighting fast response.

Here is the output of cat /etc/cron.d/pihole:

`# Pi-hole: A black hole for Internet advertisements
# (c) 2015, 2016 by Jacob Salmela
# Network-wide ad blocking via your Raspberry Pi
# http://pi-hole.net
# Updates ad sources every week
#
# Pi-hole is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 2 of the License, or
# (at your option) any later version.
#
# This file is under source-control of the Pi-hole installation and update
# scripts, any changes made to this file will be overwritten when the softare
# is updated or re-installed. Please make any changes to the appropriate crontab
# or other cron file snippets.

# Pi-hole: Update the ad sources once a week on Sunday at 01:59
#          Download any updates from the adlists
59 1    * * 7   root    PATH="$PATH:/usr/local/bin/" pihole updateGravity

# Pi-hole: Update Pi-hole! Uncomment to enable auto update
#30 2    * * 7   root    PATH="$PATH:/usr/local/bin/" pihole updatePihole

# Pi-hole: Flush the log daily at 00:00 so it doesn't get out of control
#          Stats will be viewable in the Web interface thanks to the cron job above
00 00   * * *   root    PATH="$PATH:/usr/local/bin/" pihole flush`

Does that look good? How do i know if it's firing?

**Comment by DL6ER** [comment]
From your debug log we see that your Pi-hole sends out hundreds of queries to `mail.alhaffar.home` per second coming from `192.168.1.20`. However, they are not getting answered since your Pi-hole does not know this domain.

Does this domain sound familiar to you?

**Comment by PartisanEntity** [comment]
The domain sounds familiar to me, but the mail.... doesn't sine I have not configured it.

**Comment by DL6ER** [comment]
Try closing as many programs as you can on the machine corresponding to `192.168.1.20` and watch your pihole log using `pihole -t` e.g. using SSH. Then open up program by program in order to isolate the origin of there requests - in your log it seems like they are coming up at the beginning of every minute but your recording has been to short to be absolutely sure.

**Comment by PartisanEntity** [comment]
OK thank you very much.

I ran `pihole -f` and the graphs are indeed populating now again. However I already have 4306 entries as you mentioned most going to that mail domain. I will do as you suggested.

**Comment by PartisanEntity** [comment]
Could I ask you to please anonymize the mail domain in your post? :-)

**Comment by DL6ER** [comment]
Done

**Comment by PartisanEntity** [comment]
Sorry forgot to check. Since pi-hole has a cronjob for flushing the logs, is the fact that I had so many log entries an indicator for problems with the cronjob?

**Comment by DL6ER** [comment]
No. If you generate 4300 queries per minute you will only need `1.6e6/4300/60 = 6.2` hours to generate 1.6 million requests. I suppose your computer is switched on for about this time?

**Comment by PartisanEntity** [comment]
The raspberry pi runs 24/7. So I guess in that case since I "only" have 1 million entries that is a sure indicator that the log flushing cronjob is working :-)

**Comment by DL6ER** [comment]
Agreed

**Comment by PartisanEntity** [comment]
In case anyone else runs into this here is a follow up, might help:

I had enabled RAID monitoring in webmin and had entered an email address for either webmin or mdadm (not sure which was doing this) to send an email when there is an issue with the RAID setup.

So all this time the emails were not being sent, when I looked in nullmailer queue folder there were tons of unsent emails which nullmailer was trying to send all this time.

I disable the email sending feature for RAID in webmin and also deleted all the queued emails in the nullmailer folder.

All quiet now.
